Reformat source code for consistent style

koszti opened the pull request:
<!-- Provide an overview for maintainers and reviewers. -->
## Description

This PR introduces standardized code formatting by adding `.prettierrc` and `.eslintrc.cjs` files, using the same style rules and npm targets (such as `check`, `lint`) as the main Trino Preview UI codebase. This ensures consistent code style across both codebases.

<!-- Provide details that help an engineer who is unfamiliar with this part of the code. -->
## Additional context and related issues

All existing files have been reformatted to align with the new style rules. This PR contains only formatting changes and **no new functionality or behavioral changes are introduced.**

mosabua commented:
Thoughts @nineinchnick @wendigo @georgewfisher ?

wendigo reviewed (approved):
I've reviewed configuration for prettier + workflow + ignore files.

I'd extract the automatic changes to a separate commit.

koszti commented:
@mosabua added instructions to `README.md` and updated commit message

@wendigo I’ve split the updates into two separate commits:
 1. eslint and prettier configuration, plus the README update.
 2. auto code formatting changes and enabling GHA auto check

wdyt, can this be merged?

mosabua commented:
Looks good to me. Merging.
